server/route/user.go

```diff
@@ -8,8 +8,11 @@ import (
 
 // GetCurrentUser return the current user
 func GetCurrentUser(c *gin.Context) {
-	c.JSON(http.StatusOK, gin.H{
-		"message": "GetCurrentUser",
-		"user":    "yamada",
-	})
+	user, exists := c.Get("currentUser")
```

Comment:
Don't use bare `c.Get("currentUser")`.
This leads to... well, IT IS happening now... It always becomes `exists = false` as you set with `CurrentUser` and get by `currentUser`.

Create `middleware.GetCurrentUser(c)`.